cyclus/cpp_cyclus.pxd

```diff
@@ -281,13 +330,416 @@ cdef extern from "sim_init.h" namespace "cyclus":
+        vector[std_string] GetTreePrintOuts(Agent*)
+        cpp_bool InFamilyTree(Agent*)
+        cpp_bool AncestorOf(Agent*)
+        cpp_bool DecendentOf(Agent*)
+        void Build(Agent*)
+        void EnterNotify()
+        void BuildNotify(Agent*)
+        void DecomNotify(Agent*)
+        void Decommission()
+        void AdjustMatlPrefs(PrefMap[Material].type&)
+        void AdjustProductPrefs(PrefMap[Product].type&)
+        std_string schema()
         cpp_jsoncpp.Value annotations() except +
+        const std_string prototype()
+        void prototype(std_string)
+        std_string spec()
+        void spec(std_string)
+        const std_string kind()
+        Context* context()
+        std_string str()
+        Agent* parent()
+        const int parent_id()
+        const int enter_time()
+        void lifetime(int)
+        const int lifetime()
+        const int exit_time()
+        const set[Agent*]& children()
+
+
+cdef extern from "time_listener.h" namespace "cyclus":
+
+    cdef cppclass TimeListener(Ider):
+        void Tick()
+        void Tock()
+
+
+cdef extern from "trade.h" namespace "cyclus":
+
+    cdef cppclass Trade[T]:
+        Trade()
+        Trade(Request[T]*, Bid[T]*, double)
+        Request[T]* request
+        Bid[T]* bid
+        double amt
+        double price
+
+
+
+cdef extern from "trader.h" namespace "cyclus":
+
+    cdef cppclass Trader:
+        Trader(Agent*)
+        Agent* manager()
+        set[RequestPortfolio[Material].Ptr] GetMatlRequests()
+        set[RequestPortfolio[Product].Ptr] GetProductRequests()
+        set[BidPortfolio[Material].Ptr] GetMatlBids(CommodMap[Material].type&)
+        set[BidPortfolio[Product].Ptr] GetProductBids(CommodMap[Product].type&)
+        void AdjustMatlPrefs(PrefMap[Material].type&)
+        void AdjustProductPrefs(PrefMap[Product].type&)
+        void GetMatlTrades(const vector[Trade[Material]]&,
+                           vector[pair[Trade[Material], Material.Ptr]]&)
+        void GetProductTrades(const vector[Trade[Product]]&,
+                              vector[pair[Trade[Product], Product.Ptr]]&)
+        void AcceptMatlTrades(const vector[pair[Trade[Material], Material.Ptr]]&)
+        void AcceptProductTrades(const vector[pair[Trade[Product], Product.Ptr]]&)
+
+cdef extern from "region.h" namespace "cyclus":
+
+    cdef cppclass Region(Agent, TimeListener):
+        Region(Context*)
+        void InitFromAgent "InitFrom" (Region*)
+        void InitFrom(QueryableBackend*)
+        void InfileToDb(InfileTree*, DbInit)
+        void Build(Agent*)
+        void EnterNotify()
+
+
+cdef extern from "institution.h" namespace "cyclus":
+
+    cdef cppclass Institution(Agent, TimeListener):
+        Institution(Context*)
+        void InitFromAgent "InitFrom" (Institution*)
+        void InitFrom(QueryableBackend*)
+        void InfileToDb(InfileTree*, DbInit)
+        void Build(Agent*)
+        void EnterNotify()
+        void Tock()
```

Comment:
Technically not since it inherits and tick() is pure virtual on cyclus::Institution. However, insts do have a concrete implementation of tock() that we need to call from the shim.  I'll add this for completeness, though.